packages/terra-functional-testing/lib/services/wdio-docker-service.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable class-methods-use-this */
+const path = require('path');
+const util = require('util');
+const childProcess = require('child_process');
+const Logger = require('../logger/logger');
+
+const exec = util.promisify(childProcess.exec);
+const logger = new Logger({ prefix: 'wdio-docker-service' });
+
+const RETRY_COUNT = 30;
+const POLL_INTERVAL = 2000;
+
+class DockerService {
+  /**
+   * Prepares the docker testing environment.
+   */
+  async onPrepare(config) {
+    this.host = config.hostname;
+    this.port = config.port;
+
+    await this.initializeSwarm();
+    await this.deployStack();
+  }
+
+  /**
+   * Initializes a docker swarm instance.
+   */
+  async initializeSwarm() {
+    const { stdout: dockerInfo } = await exec('docker info --format "{{json .}}"');
+    const { Swarm } = JSON.parse(dockerInfo);
+
+    if (Swarm.LocalNodeState !== 'active') {
+      logger.log('Initializing docker swarm.');
+
+      await exec('docker swarm init');
+    }
+  }
+
+  /**
+   * Deploys the docker stack.
+   */
+  async deployStack() {
+    // Remove the previous stack if one exists.
+    await this.removeStack();
+
+    logger.log('Deploying docker stack.');
+
+    const composeFilePath = path.resolve(__dirname, '../docker/docker-compose.yml');
```

Comment:
For windows developers, we will want to use path.join. 
```suggestion
    const composeFilePath = path.resolve(__dirname, path.join('..', 'docker', 'docker-compose.yml'));
```

Reply:
I'd like to get a few more opinions before making this change. I'm hesitant about developing packages that `might` work on windows. If this is all that is required for this package to fully support windows I think it would be a good addition, but if there are other issues with Windows compatibility I think we should resolve them or document our limited OS support.

Do you have any thoughts opinions @ryanthemanuel @mjhenkes  ?

Reply:
we don't support windows development

Reply:
spoke offline. Will keep as is.